src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java

```diff
@@ -150,6 +151,15 @@ public void setEnvVars(List<ContainerEnvVar> envVars) {
         this.envVars.addAll(envVars);
     }
 
+    public List<PortMapping> getPorts() {
+        return ports != null ? ports : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setPorts(List<PortMapping> ports) {
+        this.ports.addAll(ports);
```

Comment:
it's only on upgrades, maybe `envVars` was there from the beginning 

Because `ports` didn't exist, when deserializing the previous version configuration jenkins will set `ports=null`